### PR TITLE
[Android 11+] Correctly open URLs in browser and fix opening downloads and external players

### LINKED ISSUE #9771: [Android 11+] Cannot open downloads from the app ("No app installed to play this file" toast)
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

v0.25.0

### Steps to reproduce the bug

1. Download any audio from a youtube video in either .m4a or opus
2. Within the Downloads page tap on the downloaded audio file
3. A notification appears saying 'No app installed to play this file'

### Expected behavior

File should play in assigned player or a notification should appear for a suitable audio player to be chosen

### Actual behavior

A notification appears saying 'No app installed to play this file'

### Screenshots/Screen recordings

Screenshot_2023-02-09-17-24-01-97_ee8807d2de22f1831269d80920e33a3b

### Logs

_No response_

### Affected Android/Custom ROM version

Android 12

### Affected device model

Oneplus 10 Plus 5G

### Additional information

In v0.24.1 everything works as expected, it was only when I upgraded to v0.25 that the problem manifests.

I tried installing and uninstalled v0.24.1 and v0.25 a numvber of time to test the issue, and I was able to reproduce it every time.

### LINKED ISSUE #9854: [Android 11+] Cannot use external player
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.25.0

### Steps to reproduce the bug

1.Navigated to settings and then video and audio subsection.
2.Scrolled down to use external player option.
3.Upon checking it,it gets enabled.
4.Now upon opening a video in any of the resolution,a dialogue box pops in saying that no stream player has been found and prompts to install vlc.

### Expected behavior

In the earlier versions,it worked well and appropriately opened the media player and streamed the video very well however in the latest version, it's broken.

### Actual behavior

Despite having two different media player,it doesn't work anymore 

### Screenshots/Screen recordings

_No response_

### Logs

_No response_

### Affected Android/Custom ROM version

Android 11

### Affected device model

Oppo A9 2020

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Stypox**
#### What is it?
- [x] Bugfix (user facing)
- [ ] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- #9771 was caused by a misconfiguration of the `queries` field in the manifest. I thought I could use `http|https|market` to specify three schemas at once, but actually it's not possible. Hence, the permission was not actually given by the Android system and `getDefaultAppPackageName` would not work. Now I removed `https` and `market` as we never actually query those packages, but we only query `http` in `openUrlInBrowser`.
- Instead of passing a `boolean` parameter to `openUrlInBrowser`, now there are two functions:
  - `openUrlInBrowser`, which enforces the opened app is a browser
    - Remove `getDefaultAppPackageName` and just hardcode the one important line of code inside `openUrlInBrowser`. This is to make sure that function is not used by accident anywhere else, since we don't want `resolveActivity` checks anywhere (`tryOpenIntentInApp` should be used instead, **which uses a `try catch` instead**).
  - `openUrlInApp`, which allows any app to be opened
- Instead of passing a `boolean` parameter to decide whether `openIntentInApp` should show a toast, now there are two functions: `openIntentInApp` and `tryOpenIntentInApp` 
- Move all the "Kore"-related code into `KoreUtils` and deduplicate code for trying to open a Kore intent and showing a failure toast.
- Remove the `toast_no_player` string ("No app installed to play this file"), because it was basically a duplicate of `no_app_to_open_intent` ("No app on your device can open this").

For more info check out the javadocs in `ShareUtils`

#### Fixes the following issue(s)

- Fixes #9771 
- Fixes #9854
- Supersedes #9817 and #9833

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/util/external_communication/KoreUtils.java [hunk 1]
```diff
@@ -29,13 +33,39 @@ public static boolean shouldShowPlayWithKodi(@NonNull final Context context,
...
+     * <p>
+     * For a list of supported urls see the
+     * <a href="https://github.com/xbmc/Kore/blob/master/app/src/main/AndroidManifest.xml">
+     * Kore source code
+     * </a>.
+     *
+     * @param context  the context to use
+     * @param videoURL the url to the video
+     */
+    public static void playWithKore(final Context context, final Uri videoURL) {
```
Comment (AudricV) [inline app/src/main/java/org/schabi/newpipe/util/external_communication/KoreUtils.java:57]: You should keep camel case style for method parameters.

Also, Kore supports SoundCloud links, so `videoUrl` should be renamed to something like `streamUrl`:

```suggestion
    public static void playWithKore(final Context context, final Uri streamUrl) {
```

File: app/src/main/java/org/schabi/newpipe/util/external_communication/KoreUtils.java [hunk 2]
```diff
@@ -29,13 +33,39 @@ public static boolean shouldShowPlayWithKodi(@NonNull final Context context,
...
+        final Intent intent = new Intent(Intent.ACTION_VIEW)
+                .setPackage(context.getString(R.string.kore_package))
+                .setData(videoURL)
```
Comment (AudricV) [inline app/src/main/java/org/schabi/newpipe/util/external_communication/KoreUtils.java:60]:
```suggestion
                .setData(streamUrl)
```

File: app/src/main/java/org/schabi/newpipe/util/external_communication/ShareUtils.java [hunk 3]
```diff
@@ -41,122 +41,111 @@ private ShareUtils() {
...
-        final String defaultPackageName;
+     * @param context the context to use
+     * @param url     the url to browse
+     **/
+    public static void openUrlInBrowser(@NonNull final Context context, final String url) {
+        // Resolve using a generic http://, so we are sure to get a browser and not e.g. the yt app.
+        // Note that this requires the `http` schema to be added to `<queries>` in the manifest.
+        final ResolveInfo defaultBrowserInfo = context.getPackageManager().resolveActivity(
+                new Intent(Intent.ACTION_VIEW, Uri.parse("http://")),
+                PackageManager.MATCH_DEFAULT_ONLY);
```
Comment (AudricV) [inline app/src/main/java/org/schabi/newpipe/util/external_communication/ShareUtils.java:77]: Could you solve the deprecation on Android 13 and above, by using the relevant replacement on these Android versions?

**Review by AudricV — CHANGES_REQUESTED** [review]
Changes looks good overall to me, except the following ones which needs to be solved.

Please split the sentences in Javadocs into "visible" multiple lines, i.e. lines that can be seen in the Javadoc preview, especially in the ones you changed in `ShareUtils`. This would make their reading a lot better.

Could you also reword your commit message to summarize better what you have done here, so we are not really forced to look at the pull request in the future to understand better your changes? Thanks in advance!

File: app/src/main/java/org/schabi/newpipe/util/external_communication/ShareUtils.java (hunk 3 above)
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/util/external_communication/ShareUtils.java:77]: I was not sure about whether it was really important, considering that we still need to keep the deprecated method, but done

File: app/src/main/java/org/schabi/newpipe/util/external_communication/KoreUtils.java (hunk 1 above)
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/util/external_communication/KoreUtils.java:57]: I just moved this code, but you're right

**Comment by Stypox** [comment]
@AudricV done, thanks for the review :-)

**Review by AudricV — APPROVED** [review]

**Pull request merged**